### ISSUE TITLE
regex isn't consistent in python2/python3 for non-standard unicode word characters

### ISSUE THREAD

**Issue by calebrottman** [nltk/nltk#2445]
I noticed this issue when running the following command in python2 and python3:
```
nltk.pos_tag([u'açai'])
```
In python2, it is categorized as `NN` whereas in python3 it is categorized as `NNP`.

After doing some digging, i noticed that the featureset is different for these two cases. In python2, the feature "shape" is `other`. In python3, the feature "shape" is `mixedcase` (which explains the proper noun labeling).

The word "shape" is assigned in`nltk/tag/sequential.py:ClassifierPasedPOSTagger`, where we have
```
        elif re.match('\w+$', word):
            shape = 'mixedcase'
```
The `\w` matches all Unicode word characters in python3, but only `[a-zA-Z0-9_]` in python2.

I believe changing these regex calls to use the `ASCII` flag (e.g. `re.match('\w+$', word, re.ASCII)` will make the code consistent between python2 and python3.

**Comment by anantvir** [comment]
I cannot replicate the issue. When I run nltk.pos_tag([u'açai']) in Python2 and Python3 I get NN as the output.
Can you please check again with a different input and if the issue still exists, post back here with more details ?

**Comment by calebrottman** [comment]
I tried this again at home and I got NN for both python2 and 3, using nltk 3.4.5. At work I _was_ able to reproduce the NN/NNP problem, but that was on nltk 3.0.3.

FWIW - it does seem that the same problem exists with `ClassifierBasedPOSTagger` in 3.4.5, but that `pos_tag` uses that class in older versions but not in newer versions. Since I can't seem to reproduce this error in 3.4.5, I think the issue is closed. However, if you know of a place in the nltk code that still uses the `ClassifierBasedPOSTagger`, this bug may still exist.

Just FYI on the re change though:

```
python3.7 -c "import re; print(re.match('\w+$', u'açai'))"

<re.Match object; span=(0, 4), match='açai'>

python2.7 -c "import re; print(re.match('\w+$', u'açai'))"

None
```
